sdk/eventhub/azure-eventhub/samples/sync_samples/authenticate_with_sas_token.py

```diff
@@ -14,34 +14,35 @@
 import hmac
 import hashlib
 import base64
-try:
-    from urllib.parse import quote as url_parse_quote
-except ImportError:
-    from urllib import pathname2url as url_parse_quote
+from urllib.parse import quote as url_parse_quote
+from typing import TYPE_CHECKING
+
 
 from azure.core.credentials import AccessToken
 from azure.eventhub import EventHubProducerClient, EventData
+if TYPE_CHECKING:
+    from azure.eventhub import EventDataBatch
 
 
-def generate_sas_token(uri, sas_name, sas_value, token_ttl):
+def generate_sas_token(uri: str, sas_name: str, sas_value: str, token_ttl: int) -> str:
     """Performs the signing and encoding needed to generate a sas token from a sas key."""
     sas = sas_value.encode('utf-8')
     expiry = str(int(time.time() + token_ttl))
     string_to_sign = (uri + '\n' + expiry).encode('utf-8')
     signed_hmac_sha256 = hmac.HMAC(sas, string_to_sign, hashlib.sha256)
     signature = url_parse_quote(base64.b64encode(signed_hmac_sha256.digest()))
-    return 'SharedAccessSignature sr={}&sig={}&se={}&skn={}'.format(uri, signature, expiry, sas_name)
+    return f'SharedAccessSignature sr={uri}&sig={signature}&se={expiry}&skn={sas_name}'
 
 
 class CustomizedSASCredential(object):
-    def __init__(self, token, expiry):
+    def __init__(self, token: str, expiry: float) -> None:
         """
         :param str token: The token string
         :param float expiry: The epoch timestamp
 
         """
-        self.token = token
-        self.expiry = expiry
+        self.token: str = token
+        self.expiry: float = expiry
```

Comment:
don't need these annotated since they're already annotated at the parameter level